scopes/toolbox/regex/node-modules-exclude-packages/node-modules-exclude-packages.spec.ts

```diff
@@ -0,0 +1,22 @@
+import { nodeModulesExcludePackages } from './node-modules-exclude-packages';
+
+describe('nodeModulesExcludePackages()', () => {
+  it('should return an empty regex when the received array is empty', () => {
+    expect(nodeModulesExcludePackages({ packages: [] })).toEqual('node_modules/(?!()/)');
+  });
+  it('should return a regex when the received array is not empty', () => {
```

Comment:
improve the tests cases. you should have something like:
- [ ] should have yarn structure excluded
- [ ] should have old pnpm structure exlucded
- [ ] should have new pnpm structure excluded
- [ ] should work with more than one packages

also your tests shouldn't only test the final regex string, but actually the regex itself.
run the regex with a real path to a module and see that it worked. that's the most important stuff